File: javascript/src/models/endpointSecretOut.ts
// this file is @generated

export interface EndpointSecretOut {
  /**
   * The endpoint's verification secret.
   *
   * Format: `base64` encoded random bytes optionally prefixed with `whsec_`.
   * It is recommended to not set this and let the server generate the secret.
   */
  key: string;
}

export const EndpointSecretOutSerializer = {
  _fromJsonObject(object: any): EndpointSecretOut {
    return {
      key: object["key"],
    };
  },

  _toJsonObject(self: EndpointSecretOut): any {
    return {
      key: self.key,
    };
  },
};

File: rust/src/models/endpoint_secret_out.rs
// this file is @generated
use serde::{
    Deserialize,
    Serialize,
};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct EndpointSecretOut {
    /// The endpoint's verification secret.
    ///
    /// Format: `base64` encoded random bytes optionally prefixed with `whsec_`.
    /// It is recommended to not set this and let the server generate the
    /// secret.
    pub key: String,
}

impl EndpointSecretOut {
    pub fn new(key: String) -> Self {
        Self {
            key,
        }
    }
}
